featuregates/client/pkg/util/featuregate.go

```diff
@@ -131,3 +140,71 @@ func FeaturesActivatedInNamespacesMatchingSelector(ctx context.Context, c client
 	}
 	return true, nil
 }
+
+// IsFeatureActivated returns true only if the feature is activated.
+func IsFeatureActivated(ctx context.Context, c client.Client, featureName string) (bool, error) {
+	feature := &corev1alpha2.Feature{}
+	if err := c.Get(ctx, types.NamespacedName{
+		Name: featureName,
+	}, feature); err != nil {
+		return false, fmt.Errorf("couldn't retrieve feature %s :%w", featureName, err)
+	}
+
+	return feature.Status.Activated, nil
+}
+
+// GetFeatureGateForFeature returns FeatureGate resource that is gating the feature
+func GetFeatureGateForFeature(ctx context.Context, c client.Client, featureName string) (*corev1alpha2.FeatureGate, bool, error) {
```

Comment:
Not important right away, but can we [setup FieldIndexers](https://github.com/kubernetes-sigs/controller-runtime/blob/4c9c9564e4652bbdec14a602d6196d8622500b51/pkg/client/example_test.go#L249-L268) to answer these types of "contains" queries with less cost? These functions are used the in reconcile loop meaning there will be a lot of GET API calls.

Reply:
ah right.. i totally forgot about FieldIndexers, i think this makes sense.. i will do this in a follow up pr

Reply:
I remember why I did this now, I was thinking from a perspective that these functions could be helpful on the client side as well. Will revisit this comment later

Reply:
@yharish991 , do we have an GH issue or some other tracking system for this ? If not, let's create one. Thanks

Reply:
i created an issue for this in our internal tracking system